tests/kernel/profiling/profiling_api/src/main.c

```diff
@@ -90,7 +90,11 @@ static void thread_callback(const struct k_thread *thread, void *user_data)
 
 	if (((char *)thread->stack_info.start ==
 			K_THREAD_STACK_BUFFER(test_stack)) &&
+#if defined(CONFIG_USERSPACE) && defined(CONFIG_ARC)
+			(thread->stack_info.size >= STACKSIZE)) {
```

Comment:
This isn't right.

On other arches, and all the code assumes:

K_THREAD_STACK_BUFFER should be the lowest memory address of the actual writable region of the stack, the actual stack buffer itself. It should equal thread->stack_info.start.
thread->stack_info.size should indicate the bounds of the actual writable stack region.
These changes shouldn't be necessary.